node/node.go

```diff
@@ -286,6 +254,8 @@ func logNewNodeInfo(state sm.State, privValidator types.PrivValidator, logger, c
 	}
 }
 
+// This creates and starts the private validator socket client if, and only if,
+// `config.PrivValidatorListenAddr` is non-empty.
 func optionallyCreateAndStartPrivValidator(config *cfg.Config, privValidator types.PrivValidator, logger log.Logger) (types.PrivValidator, error) {
 	if config.PrivValidatorListenAddr != "" {
```

Comment:
early return?

```go
if config.PrivValidatorListenAddr == "" {
  return nil, nil
}
```

Reply:
We can return early here like this (which effectively results in a pass-through of whichever `privValidator` was passed to us):

```go
if config.PrivValidatorListenAddr == "" {
    return privValidator, nil
}
```